api/src/test/java/org/asynchttpclient/test/TestUtils.java

```diff
@@ -92,6 +99,57 @@ public static File createTempFile(int approxSize) throws IOException {
         }
     }
 
+    public static class SyncByteBufferPublisher implements Publisher<ByteBuffer> {
+        // FIXME If I reduce the chunk size the test will fail. Why?
+        private static final int CHUNK_SIZE = 30000;
+
+        private final byte[] bytes; // payload streamed to the subscriber
+        private final int chunkSize; // number of bytes in each element sent to
+                                     // the subscriber
+
+        public SyncByteBufferPublisher(byte[] bytes) {
+            this(bytes, CHUNK_SIZE);
+        }
+
+        public SyncByteBufferPublisher(byte[] bytes, int chunkSize) {
+            this.bytes = Arrays.copyOf(bytes, bytes.length);
+            this.chunkSize = chunkSize;
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super ByteBuffer> subscriber) {
+            subscriber.onSubscribe(new Subscription() {
+                // This index is used to keep track of bytes that have been sent.
+                private volatile int currentIndex = 0;
+                // This flag will track whether this `Subscription` is to be considered cancelled or not
+                private volatile boolean cancelled = false;
+
+                public void request(long n) {
```

Comment:
This will need to be modified to avoid blowing the stack in case `request` is invoked by `onNext` - the reactive stream spec requires that this be handled here, not in `Subscriber.onNext`.  A better option here might be to pull in a reactive streams implementation for the tests, eg rxjava or akka streams. @slandelle do you have a preference as to which implementation should be used?